### DISCUSSION TITLE
How would we classify overly permissive Github action

### DISCUSSION THREAD

**Discussion by rubtoa** [pbom-dev/OSCAR#36] [General]
So a Github action can be assigned read/write permissions for the following scopes
actions
  checks
  contents
  deployments
  id-token
  issues
  discussions
  packages
  pages
  pull-requests
  repository-projects
  security-events
  statuses

So the attack step technique can vary - my thoughts are that it's either "Lateral Movement" or "Persisance"

**Comment by vaq130** [comment]
I am not sure if this question matches the GitHub permissions model:  "So the attack step technique can vary - my thoughts are that it's either "Lateral Movement" or "Persisance""

Could you restate the issue?

**Comment by rubtoa** [comment]
Yes, sorry i was kinda thinking out loud :)

The issue i want to discuss is the abuse of overly-permissive Github actions workflow - ("permissions: write-all"). If an attacker can control the execution code - it is possible to do a lot of bad stuff. Planting malware code in the repo or altering artifacts for instance.
1. Does it fall into an existing technique? (i think not)
2. If we will add a technique under which tactic will we put it?

**Comment by vaq130** [comment]
Thanks for clarifying.  Yes, this is a very important tactic to include and I believe it does fall under Persistence.

persistance - The adversary is trying to maintain their foothold.

**Comment by rubtoa** [comment]
Added a task #42
